android/src/main/java/com/appgoalz/rnjwplayer/RNJWPlayerView.java

```diff
@@ -323,6 +325,17 @@ public AppViewFullscreenHandler(RNJWPlayer player) {
 
         @Override
         public void onFullscreenRequested() {
+            // prevent a race condition from occurring when the player is initialized
+            // and the android back button is pressed immediately after
+            // this would normally call onFullscreenExitRequested first and then onFullscreenRequested,
+            // which would cause the app to crash
+            // see https://nuuvuu.atlassian.net/browse/FSB-1466
+            if (hasRequestedExitFullscreen) {
+                hasRequestedExitFullscreen = false;
+                return;
+            }
+            isFullscreen = true;
```

Comment:
tu nie je ziaden dovod assumovat ze to je synchrony kod. Toto musi byt but atomic alebo sychronized